src/sql/base/browser/ui/table/table.ts

```diff
@@ -405,4 +463,59 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 	public get container(): HTMLElement {
 		return this._tableContainer;
 	}
+
+	// Checks if the current table is a tree grid. Every tree grid should have a column with expandableColumnFormatter
+	private isTreeGrid(): boolean {
+		return this._grid.getColumns().filter(c => c.formatter === treeGridExpandableColumnFormatter).length > 0;
+	}
+
+	// Gets the index for the expandable column
+	private expandableColumnIndex(): number {
+		return this._grid.getColumns().findIndex(c => c.formatter === treeGridExpandableColumnFormatter);
+	}
+
+	// This expands/collapse the tree grid item
+	private toggleTreeGridParent(row: number, cell: number, forceState?: boolean): void {
+		const rowData = this._data.getItem(row);
+		if (rowData['isParent'] && this._grid.getColumns()[cell].formatter === treeGridExpandableColumnFormatter) {
+			if (forceState === undefined) {
+				if (!rowData.expanded) {
+					(<any>rowData).expanded = true;
+				} else {
+					(<any>rowData).expanded = false;
+				}
+			} else {
+				(<any>rowData).expanded = forceState;
+			}
+			this._data.filter(this._grid.getColumns());
+			this.rerenderGrid();
+			this.focus();
+		}
+	}
+
+	// We need to transform grid data to tree grid data. This includes adding required aria attributes to the rows
+	private transGridDataToTreeGridData(data: IDisposableDataProvider<T>): IDisposableDataProvider<T> {
+		for (let i = 0; i < data.getLength(); i++) {
+			const dataRow = <any>data.getItem(i);
```

Comment:
This should be T - why are you casting it to any? If the type isn't right we should be fixing that instead